Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -624,12 +630,45 @@ void CGPathApply(CGPathRef path, void* info, CGPathApplierFunction function) {
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 bool CGPathContainsPoint(CGPathRef path, const CGAffineTransform* m, CGPoint point, bool eoFill) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (m) {
+        point = CGPointApplyAffineTransform(point, *m);
+    }
+    // check if the point is outside this box already, if it is, return false
+    CGRect boundingBox = CGPathGetBoundingBox(path);
+
+    if ((boundingBox.origin.x > point.x) || (boundingBox.origin.y > point.y) ||
+        ((boundingBox.origin.x + boundingBox.size.width) < point.x) || ((boundingBox.origin.y + boundingBox.size.height) < point.y)) {
+        return false;
+    }
+
+    CGContextRef context =
+        CGBitmapContextCreate(0, boundingBox.origin.x + boundingBox.size.width, boundingBox.origin.y + boundingBox.size.height, 1, 1, 0, 0);
+
+    CGContextSetGrayFillColor(context, 1.0, 1.0);
+    CGContextAddPath(context, path);
+
+    if (eoFill) {
+        CGContextEOFillPath(context);
+    } else {
+        CGContextFillPath(context);
+    }
+
+    CGImageRef image = context->Backing()->DestImage();
+
+    float r = 0.0f;
+    float g = 0.0f;
+    float b = 0.0f;
+    float a = 0.0f;
+    image->Backing()->GetPixel(point.x, point.y, r, g, b, a);
```

Comment:
@jaredhms - I implemented the below in CGContextCairo (see code sample below), and both the CGCatalog sample app and my unit tests fail in the rounded corners. I noticed in the cairo documentation that clipping is not counted, and that may be the problem here. Do you prefer having the current PR implementation that mimics iOS more closely? Or do you prefer the behavior cairo's implementation? 

`bool CGContextCairo::CGContextIsPointInPath(bool eoFill, float x, float y) {
    ObtainLock();

    LOCK_CAIRO();
    if(eoFill){
        cairo_set_fill_rule(_drawContext, CAIRO_FILL_RULE_WINDING);
        } else {
        cairo_set_fill_rule(_drawContext, CAIRO_FILL_RULE_EVEN_ODD);
        }
    bool returnValue = cairo_in_fill(_drawContext, x, y);
    UNLOCK_CAIRO();
    return returnValue;
}`